tests/phpunit/tests/block-template-utils.php

```diff
@@ -86,7 +86,12 @@ public function set_up() {
 		switch_theme( self::TEST_THEME );
 	}
 
-	public function test_build_block_template_result_from_post() {
+	/**
+	 * @ticket 54335
+	 *
+	 * @covers ::_build_block_template_result_from_post
+	 */
+	public function test_build_block_template_result_from_post_template() {
```

Comment:
The name's a bit hard to read; how about we keep this as it was before

```suggestion
	public function test_build_block_template_result_from_post() {
```

and change the new one from `test_build_block_template_result_from_post_template_part` to `test_build_block_template_part_result_from_post_template`?

I realize that would mean that it's no longer following the `test_{function_name}_xyz` pattern, but maybe that's okay, given that we have the `@covers` PHPDoc to indicate the function it's covering.

Reply:
What about moving the respective tests to their own files and using the `test_should_` / `test_should_not_` format ([Ref](https://make.wordpress.org/core/handbook/testing/automated-testing/writing-phpunit-tests/#test-methods)) for test names, which also reduces redundant references to the function's name?

i.e.

Filename: `buildBlockTemplateResultFromPost.php`
Classname: `Tests_Block_Template_Utils_BuildBlockTemplateResultFromPost`
Tests:
  - `test_should_build_template()`
  - `test_should_build_template_part()`
---
Filename: `buildBlockTemplateResultFromFile.php`
Classname: `Tests_Block_Template_Utils_BuildBlockTemplateResultFromFile`
Tests:
  - `test_should_build_template()`
  - `test_should_build_template_using_custom_properties()`
  - `test_should_enforce_default_properties_when_building_template()`
  - `test_should_respect_post_types_property_when_building_template()`
  - `test_should_build_template_part()`
  - `test_should_ignore_post_types_property_when_building_template_part()`

Using this approach may also help with identifying possible additional tests.

Reply:
I refactored the code as recommended with f0250e35660ab87bd2c78c497908365a8d9d278d. Let me know if it that works for you.